lib/jxl/encode.cc

```diff
@@ -418,6 +427,24 @@ void JxlEncoderInitBasicInfo(JxlBasicInfo* info) {
   info->animation.have_timecodes = JXL_FALSE;
 }
 
+void JxlEncoderInitFrameHeader(JxlFrameHeader* info) {
+  info->duration = 0;
+  info->timecode = 0;
+  info->name_length = 0;
+  info->is_last = JXL_FALSE;
+  info->layer_info.crop_x0 = 0;
+  info->layer_info.crop_y0 = 0;
+  // TODO(lode): must these be set to image size instead, or will caller do
+  // this?
+  info->layer_info.xsize = 0;
+  info->layer_info.ysize = 0;
+  info->layer_info.blend_info.blendmode = JXL_BLEND_REPLACE;
+  info->layer_info.blend_info.source = 0;
+  info->layer_info.blend_info.alpha = 0;
+  info->layer_info.blend_info.clamp = 0;
+  info->layer_info.save_as_reference = 0;
```

Comment:
If zero duration layers are the default, then I guess source and save_as_reference should both be 1, right @veluca93 ?

If kReplace is the default and full-sized frames are the default, then it's fine to keep these at 0.

Reply:
I'd say correct.

Reply:
See other comment: I chose JXL spec default for everything, and the default are 0. User should set up values when needed for the user's purpose